components/datadog/agentparams/params.go

```diff
@@ -117,7 +120,19 @@ func WithAgentConfig(config string) func(*Params) error {
 // WithIntegration adds the configuration for an integration.
 func WithIntegration(folderName string, content string) func(*Params) error {
 	return func(p *Params) error {
-		p.Integrations[folderName] = content
+		confPath := path.Join("conf.d", folderName, "conf.yaml")
+		p.Integrations[confPath] = content
+		return nil
+	}
+}
+
+// WithFile adds a file to the install with the contents at the given path
+func WithFile(filePath string, content string) func(*Params) error {
+	return func(p *Params) error {
+		if !strings.HasPrefix(filePath, "/") && !strings.HasPrefix(filePath, "C:\\") {
```

Comment:
I'm not sure if `C:\` is the correct windows install partition for the agent but I assume so?

Reply:
👍 awesome! good catch

Reply:
hmm on closer look I don't think I can use this as it references the runtime.GOOS to determine what is a valid path for the runtime architecture but we are referencing a VM that could be different from the CI box the test is running on

Reply:
Indeed you cannot use `IsAbs`. I didn't realise the path was the remote path.

Reply:
Fixed in https://github.com/DataDog/test-infra-definitions/pull/289/commits/febd9a54ac2670ee61776df1c4a658bc9857d48c